parser/prism/Translator.cc

```diff
@@ -1866,7 +1866,18 @@ template <typename PrismNode> std::unique_ptr<parser::Mlhs> Translator::translat
                 auto location = translateLoc(splatNode->base.location);
                 auto expr = translate(splatNode->expression);
 
-                sorbetLhs.emplace_back(make_unique<parser::SplatLhs>(location, move(expr)));
+                std::unique_ptr<parser::Node> sorbetPlatNode;
+
+                auto argNode = dynamic_cast<parser::Arg *>(expr.get());
```

Comment:
I'd like to avoid doing dynamic type checks if possible. I'm thinking: rather than checking the type of the result on the way out of `translate()`, can we instead check something about the input before `translate()` instead?

From what I can tell, `translate()` can only return a `pasrser::Arg` in the `case PM_REQUIRED_PARAMETER_NODE`, so we do a (faster) check for that instead:

```c++
auto expr = translate(splatNode->expression);

if (PM_NODE_TYPE_P(splatNode->expression, PM_REQUIRED_PARAMETER_NODE)) {
    sorbetLhs.emplace_back(make_unique<parser::SplatLhs>(location, move(expr));
} else {
    sorbetLhs.emplace_back(make_unique<parser::Restarg>(location, argNode->name, argNode->loc));
}
```

WDYT?

Some added comments to explain the reasoning behind the `if`/`else` would be nice, too.